pufferlib/ocean/trash_pickup/trash_pickup.c

```diff
@@ -0,0 +1,74 @@
+#include <time.h>
+#include "trash_pickup.h"
+
+#define INDEX_C(env, x, y) ((y) * (env).grid_size + (x))
+
+// Demo function for visualizing the TrashPickupEnv
+
+
+void demo(int grid_size, int num_agents, int num_trash, int num_bins, int max_steps) {
+    CTrashPickupEnv env;
+    env.grid_size = grid_size;
+    env.num_agents = num_agents;
+    env.num_trash = num_trash;
+    env.num_bins = num_bins;
+    env.max_steps = max_steps;
+
+    allocate(&env);
+
+    Client* client = make_client(&env);
+
+    reset(&env);
+
+    while (!WindowShouldClose()) {
+        // Random actions for all agents
+        for (int i = 0; i < env.num_agents; i++) {
+            env.actions[i] = rand() % 4; // 0 = UP, 1 = DOWN, 2 = LEFT, 3 = RIGHT
+            // printf("action: %d \n", env.actions[i]);
+        }
+
+        // Step the environment and render the grid
+        step(&env);
+        render(client, &env);
+    }
+
+    free_allocated(&env);
+    close_client(client);
+}
+
+
+
+// Performance test function for benchmarking
+void performance_test() {
+    long test_time = 10; // Test duration in seconds
+
+    CTrashPickupEnv env;
+    env.grid_size = 3;
+    env.num_agents = 1;
```

Comment:
For multiagent envs, do your test with realistic conditions (multiple agents) and multiply sps by num agents, because the metric should really be agent steps / second

Reply:
Completed (to be pushed).